File: leetcode2/2medium/최은서/Q2750.js
/**
 * @param {number[]} nums
 * @return {number}
 */

// subarray : 1이 하나만 있는 부분 배열
// 1 0 1 -> 나눌 수 없음

// 1 1 0 -> 1, 1 0

// 1 0 0 1 -> 1, 0 0 1 / 1 0, 0 1 / 1 0 0, 1 : 3가지

// 0 1 0 0 1 0 1 : (2+1)가지 * (1+1)가지 = 6가지
// 0 1 , 0 0 1, 0 1 / 0 1 0, 0 1, 0 1 / 0 1 0 0, 1, 0 1
// 0 1 , 0 0 1 0, 1 / 0 1 0, 0 1 0, 1 / 0 1 0 0, 1 0, 1
var numberOfGoodSubarraySplits = function(nums) {
    let zeroCount = 0
    let isCountable = false;
    let res = 1;
    const MOD = 1000000007;
    for (const n of nums) {
        if (n === 0 && isCountable) {
            zeroCount += 1;
        }
        else if (n === 1) {
            isCountable = true
            if (isZero !== 0) {
                res = (res * (zeroCount+1)) % MOD;
                zeroCount = 0;
            }
        }
    }
    if (!isCountable) return 0; // [0, 0] -> 0
    return finalResult;
};

File: leetcode2/2medium/김도연/Q2750.java
package Leetcode.김도연;

public class Q2750 {
    public int numberOfGoodSubarraySplits(int[] nums) {
        int MOD = 1_000_000_007;
        int n = nums.length;
        int result = 1;
        int i = 0;

        // 처음 1 위치 찾기
        while (i < n && nums[i] == 0) {
            i++;
        }

        // 배열에 1이 없는 경우
        if (i == n)
            return 0;

        for (int j = i + 1; j < n; j++) {
            if (nums[j] == 1) {
                result = (int) ((long)result * (j-i) % MOD);
                i = j;
            }
        }

        return result;

    }
}
